memory/memory.go

```diff
@@ -117,34 +117,50 @@ func (q *Queue) Transaction(txcb queue.TxCallback) error {
 	return nil
 }
 
-// Consume implements Queue.  MemoryQueues have infinite advertised window.
-func (q *Queue) Consume(_ int) (queue.JobIter, error) {
-	return &JobIter{q: q, RWMutex: &q.RWMutex, finite: q.finite}, nil
+// Consume implements Queue. The advertisedWindow value is the maximum number of
+// unacknowledged jobs. Use 0 for an infinite window.
+func (q *Queue) Consume(advertisedWindow int) (queue.JobIter, error) {
+	jobIter := JobIter{
+		q:       q,
+		RWMutex: &q.RWMutex,
+		finite:  q.finite,
+	}
+
+	if advertisedWindow > 0 {
+		jobIter.sem = make(chan struct{}, advertisedWindow)
```

Comment:
Please rename it, to:
```suggestion
		jobIter.chn = make(chan struct{}, advertisedWindow)
```
with `sem` it's super confusing now.

Reply:
At least `chn` tells me something about type. In go not so often you use _semaphores_, moreover without having _semaphore_ type I don't know what to expect.
This is just a proposal - up to you.

Reply:
I don't really mind one way or the other. Changed in e25f9e5.